src/world_generation/biome.rs

```diff
@@ -35,11 +36,20 @@ pub struct BiomeConfig {
     pub noise_config: [PerlinNoiseConfig; MAX_LEVEL_NOISE],
     pub layers: [Option<BiomeLayer>; MAX_NUMBER_LAYER],
     pub num_layer: usize,
+
+    pub tree_probability: f32,
 }
 
 impl BiomeConfig {
     // The constructor enforces that layers respect the convention, allowing for efficient binary search
```

Comment:
Doc comment starts with `///` instead of `//`

Remove the "The constructor". It's a constructor, it's called `new`. Just describes what it does and panic conditions: 

```
///Enforcers that layers ....
/// Panics if : CONDITIONS
```